WcaOnRails/app/webpacker/components/Competitions/MyCompetitions/index.jsx

```diff
@@ -0,0 +1,310 @@
+          <Table.HeaderCell />
+          <Table.HeaderCell />
+        </Table.Row>
+      </TableHeader>
+
+      <TableBody>
+        {competitions.map((competition) => (
+          <Table.Row key={competition.id}>
+            <Table.Cell>
+              <a href={competition.url}>{competition.name}</a>
+            </Table.Cell>
+            <Table.Cell>
+              {cityAndCountry(competition)}
+            </Table.Cell>
+            <Table.Cell>
+              {dateRange(competition.start_date, competition.end_date)}
+            </Table.Cell>
+            <Table.Cell>
+              {!competition['results_posted?'] && (
+                <Icon name="calendar check" />
+              )}
+            </Table.Cell>
+            <Table.Cell>
+              {competition['results_posted?'] && (
+              <Popup
+                content={I18n.t('competitions.my_competitions_table.results_up')}
+                trigger={(
+                  <Icon name="check circle" />
+                )}
+              />
+              )}
+            </Table.Cell>
+            <ReportTableCell competitionId={competition.id} permissions={permissions} />
+          </Table.Row>
+        ))}
+      </TableBody>
+    </Table>
+  );
+}
+
+export default function MyCompetitions() {
+  const [isAccordionOpen, setIsAccordionOpen] = useState(false);
+  const [shouldShowRegistrationStatus, setShouldShowRegistrationStatus] = useState(false);
+
+  const { data: competitions, loading: competitionsLoading } = useLoadedData(myCompetitionsAPIUrl);
+  const { data: me, loading: meLoading } = useLoadedData(meAPIUrl);
+  const { data: permissions, loading: permissionsLoading } = useLoadedData(permissionsAPIUrl);
+
+  return (
+    (meLoading || competitionsLoading || permissionsLoading) ? <Loading /> : (
+      <>
+        <Header>
+          {I18n.t('competitions.my_competitions.title')}
+        </Header>
+        <p>
+          {I18n.t('competitions.my_competitions.disclaimer')}
+        </p>
+        <UpcomingCompetitionTable
+          competitions={competitions?.future_competitions ?? []}
+          permissions={permissions ?? defaultPermissions}
+          registrationStatuses={competitions?.registered_for_by_competition_id ?? {}}
+        />
+        <Accordion fluid styled>
+          <Accordion.Title
+            active={isAccordionOpen}
+            onClick={() => setIsAccordionOpen(!isAccordionOpen)}
+          >
+            {`${I18n.t('competitions.my_competitions.past_competitions')} (${competitions?.past_competitions?.length ?? 0})`}
+          </Accordion.Title>
+          <Accordion.Content active={isAccordionOpen}>
+            <PastCompetitionsTable
+              competitions={competitions?.past_competitions ?? []}
+              permissions={permissions ?? defaultPermissions}
+            />
+          </Accordion.Content>
+        </Accordion>
+        <a href={personUrl(me.user.wca_id)}>{I18n.t('layouts.navigation.my_results')}</a>
```

Comment:
What if `me` errors?

Reply:
Seems like we need a generic "something went wrong, please refresh" page to show when errors pop up. Do we currently have such an Error Boundary for our react pages?